apps/pipeline/lib/pipeline/writer/s3_writer.ex

```diff
@@ -173,4 +183,24 @@ defmodule Pipeline.Writer.S3Writer do
       error -> error
     end
   end
+
+  defp delete_json_table(table_name) do
+    json_table_name = table_name("JSON", table: table_name)
+
+    json_table_name
+    |> StatementUtils.create_new_table_with_existing_table()
+
+    json_table_name
+    |> StatementUtils.drop_table()
+  end
+
+  defp delete_orc_table(table_name) do
+    orc_table_name = table_name("ORC", table: table_name)
+
+    orc_table_name
+    |> StatementUtils.create_new_table_with_existing_table()
+
+    orc_table_name
+    |> StatementUtils.drop_table()
+  end
```

Comment:
These functions are exactly the same.  Consider consolidating them into one that just takes the table name as a parameter.

Reply:
Agreed